src/group_theory/commuting_probability.lean

```diff
@@ -0,0 +1,65 @@
+/-
+Copyright (c) 2022 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import algebra.group.conj
+import group_theory.group_action.conj_act
+
+/-!
+# Commuting Probability
+This file introduces the commuting probability of finite groups.
+
+## Main definitions
+* `comm_prob`: The commuting probability of a finite type with a multiplication operation.
+
+## Todo
+* Neumann's theorem.
+-/
+
+noncomputable theory
+open_locale classical
+open_locale big_operators
+
+open nat fintype
+
+variables (M : Type*) [fintype M] [has_mul M]
+
+/-- The commuting probability of a finite type with a multiplication operation -/
+def comm_prob : ℚ := card {p : M × M // p.1 * p.2 = p.2 * p.1} / card (M × M)
```

Comment:
Wouldn't it typically be more practical to divide by `(card M) ^ 2`? Because there will usually be more info about `card M` then about `card (M × M)`.

Reply:
It made a few proofs longer, but it's probably ok.